src/borg/archive.py

```diff
@@ -83,21 +83,33 @@ def __add__(self, other):
         stats.osize_parts = self.osize_parts + other.osize_parts
         stats.usize_parts = self.usize_parts + other.usize_parts
         stats.nfiles_parts = self.nfiles_parts + other.nfiles_parts
-        stats.files_stats = other.files_stats
-        stats.chunking_time = other.chunking_time
-        stats.hashing_time = other.hashing_time
+        stats.chunking_time = self.chunking_time + other.chunking_time
+        stats.hashing_time = self.hashing_time + other.hashing_time
+        stats.files_stats["A"] = self.files_stats["A"] + other.files_stats["A"]
+        stats.files_stats["U"] = self.files_stats["U"] + other.files_stats["U"]
+        stats.files_stats["M"] = self.files_stats["M"] + other.files_stats["M"]
```

Comment:
btw, i just noticed that in case there is no "added file", such code will crash, because then there will be no "A" key in the dict.

maybe it is simpler to not have that dict and just some `stats.*` attributes likes files_added, files_modified, files_unchanged.

Reply:
I think it won't because the dict is initialized with the three keys in `Statistics.__init__`.